### ISSUE TITLE
Allow `interactive: true` to force interleaved output in order to make interactive CLI apps work

### ISSUE THREAD

**Issue by stephenprater** [go-task/task#217]
I have encountered several different instances of TTY / term control escape codes not working the way that they should.

The easiest possible repro is this:

```
tasks:
    shell:
      cmd:
         - /usr/local/bin/zsh
```

![Screenshot 2019-06-17 19 07 36](https://user-images.githubusercontent.com/149870/59648267-53fd9c80-9133-11e9-9ab1-6f2aa9af114c.png)

Some terminal-centric programs will not work at all.  (`Ipython` is an example of this - which appears to run - but will not output anything until you close it with Ctrl-D)

Although I haven't looked into this in depth - I _suspect_ that this because not all terminal escape codes are forwarded correctly - or because the underlying program _checks_ to see if it's connected to a TTY and does a different thing if it's not.

For instance - this prints 1-100 in place when run under `/bin/sh`

```sh
for i in $(seq 100); do
   printf "$(tput sc) $i $(tput rc)";
done
```

When run under task it just prints:

```
[count] 100
```

Apologies for the title of this issue.  I'm not exactly sure how to describe it.

**Comment by andreynering** [comment]
Hey @stephenprater,

Thanks for opening an issue for this!

I see by the output you that seem to be using the `prefixed` output, right? Unfortunately, I think we have some bugs on running interactive commands when not using the interleaved (default) output. Can you check if you can still reproduce it using interleaved? I supposed it'll work as expected.

Unfortunately, I think this is a bug we just can't solve (how would an interactive program like Vim, etc. run while having a prefix? That don't even makes sense IMO).

I think we should probably allow adding `interactive: true` to tasks or commands, that will use the interleaved output even when something else was set.

Perhaps we should even lock when an interactive program is ran: I caught bugs in the past when running more than one interactive program in parallel:

```yaml
version: '2'

tasks:
  default:
    deps: [vim, vim] # run this and have fun with a extremely buggy Vim =)

  vim:
    cmds:
      - vim foo.txt
```

**Comment by andreynering** [comment]
Ping @stephenprater

**Comment by stephenprater** [comment]
Yes - it works without `output: prefixed` and an `interactive: true` seems like a fine solution short of providing a per task `output` config.

That said - this is less of a problem for interactive programs like `vim` and more for things like progress bars - which don't work because the TTY isn't forwarded correctly?

Like this:

```
tasks:
  count:
    cmds:
      - |
        for i in $(seq 1000); do
          printf "$(tput sc) $i $(tput rc)";
        done
```

Will count to 1000 in place without `prefixed` but just pauses and then print `1000` with prefixed.

This should be able to work.

**Comment by jonas27** [comment]
I am encountering the same (???) error. Prefixed is not printing immediately but only after task execution.

**Comment by andreynering** [comment]
I just updated the issue title with the proposal.

@jonas27 If you're using prefixed output together with interactive CLI applications, it's likely the same bug.

**Comment by SwitchedToGitlab** [comment]
Please correct me if I'm wrong here, but on version 3 of my Taskfile using VIM this works as I would expect. Same goes for the count examples above.

@andreynering is this still an issue that you believe needs to be addressed in version > 3.6.0?

**Comment by andreynering** [comment]
These problems still happen for me. The title's feature request is still valid.
